File: challenge_218/challenge_218.rb
def to_palindromic(number)
  unless number.length <= 1 or is_palindromic(number)
    return to_palindromic((number.to_i + (number.reverse).to_i).to_s)
  end
  
  return number
end

def is_palindromic(number)
  return number == number.reverse
end

number = ARGV[0]
puts(to_palindromic(number))

File: challenge_218/challenge_218.py
def to_palindromic(number, step):
    if len(number) <= 1 or step > 1000 or is_palindromic(number):
        return number
    else:
        return to_palindromic(str(int(number) + int(number[::-1])), step + 1)

def is_palindromic(number):
    return number == number[::-1]

number = input("Enter a number: ")
print(to_palindromic(number, 0))
